framework/src/processing/lib/spark-runtime/emr-containers/spark-emr-containers-runtime.ts

```diff
@@ -620,5 +678,41 @@ export class SparkEmrContainersRuntime extends TrackedConstruct {
 
     return manifestApply;
   }
+
+  /**
+   * Creates a new Amazon EMR managed endpoint to be used with Amazon EMR Virtual Cluster .
+   * CfnOutput can be customized.
+   * @param {Construct} scope the scope of the stack where managed endpoint is deployed
+   * @param {string} id the CDK id for endpoint
+   * @param {SparkEmrContainersRuntimeInteractiveSessionProps} interactiveSessionOptions the EmrManagedEndpointOptions to configure the Amazon EMR managed endpoint
+   */
+
+  public addInteractiveEndpoint(scope: Construct, id: string, interactiveSessionOptions: SparkEmrContainersRuntimeInteractiveSessionProps) {
+
+    if (interactiveSessionOptions.managedEndpointName.length > 64) {
```

Comment:
Should we add a unique hash at the end of the name? Does it need to be unique?

Reply:
no, there is no need for uniqueness, the uniqueness is handled by the physical resource id, managed_endpoint_id.